pkg/processor/runtime/python/py/_nuclio_wrapper.py

```diff
@@ -62,25 +71,76 @@ def __init__(self, logger, handler, socket_path, platform_kind, namespace=None,
+    def _event_from_msgpack(parsed_data):
+        """Decode event encoded as MessagePack by processor"""
+
+        trigger = TriggerInfo(
+            parsed_data['trigger']['class'],
+            parsed_data['trigger']['kind'],
+        )
+
+        # extract content type, needed to decode body
+        content_type = parsed_data['content_type']
+
+        body = nuclio_sdk.Event.decode_body(parsed_data['body'], content_type)
+
+        return nuclio_sdk.Event(body=body,
+                                content_type=content_type,
+                                trigger=trigger,
+                                fields=parsed_data.get('fields'),
+                                headers=parsed_data.get('headers'),
+                                _id=parsed_data['id'],
+                                method=parsed_data['method'],
+                                path=parsed_data['path'],
+                                size=parsed_data['size'],
+                                timestamp=datetime.datetime.utcfromtimestamp(parsed_data['timestamp']),
+                                url=parsed_data['url'],
+                                _type=parsed_data['type'],
+                                type_version=parsed_data['type_version'],
+                                version=parsed_data['version'])
+
     def serve_requests(self, num_requests=None):
         """Read event from socket, send out reply"""
 
+        int_buf = bytearray(8)
+        buf = memoryview(bytearray(4 * 1024 * 1024))
+
         while True:
 
             formatted_exception = None
             encoded_response = '{}'
 
             try:
-                line = self._processor_sock_rfile.readline()
 
+                should_be_eight = self._processor_sock.recv_into(int_buf, 8)
                 # client disconnect
-                if not line:
+                if should_be_eight < 8:
                     # If socket is done, we can't log
                     print('Client disconnect')
                     return
+                bytes_to_read = int.from_bytes(int_buf, "big")
+
+                cumulative_bytes_read = 0
+                while cumulative_bytes_read < bytes_to_read:
+                    view = buf[cumulative_bytes_read:bytes_to_read]
+                    bytes_to_read_now = bytes_to_read-cumulative_bytes_read
+                    bytes_read = self._processor_sock.recv_into(view, bytes_to_read_now)
+
+                    # client disconnect
+                    if bytes_read == 0:
+                        # If socket is done, we can't log
+                        print('Client disconnect')
+                        return
+
+                    cumulative_bytes_read += bytes_read
+
+                self._unpacker.feed(buf[:cumulative_bytes_read])
+
+                msg = next(self._unpacker)
 
                 # decode the JSON encoded event
-                event = nuclio_sdk.Event.from_json(line)
+                event = Wrapper._event_from_msgpack(msg)
```

Comment:
make Wrapper's `_event_from_msgpack` public if you use it from outside

Reply:
actually im wrong, check out this comment https://github.com/nuclio/nuclio/pull/1312/files#r306719882